Cython/Compiler/Nodes.py

```diff
@@ -3885,10 +3901,24 @@ def generate_keyword_unpacking_code(self, min_positional_args, max_positional_ar
             code.putln('default:')
         for i in range(max_positional_args-1, -1, -1):
             code.put('case %2d: ' % (i+1))
-            code.putln("values[%d] = PyTuple_GET_ITEM(%s, %d);" % (
-                i, Naming.args_cname, i))
-            code.putln('CYTHON_FALLTHROUGH;')
-        code.putln('case  0: break;')
+            if i+1 > num_required_posonly_args:
+                code.putln("values[%d] = PyTuple_GET_ITEM(%s, %d);" % (
+                    i, Naming.args_cname, i))
+                code.putln('CYTHON_FALLTHROUGH;')
+            elif i+1 == num_required_posonly_args:
+                for j in range(num_required_posonly_args-1, -1, -1):
```

Comment:
This looks like it could be done after the loop, by running the outer loop only up to `num_required_posonly_args` instead of `-1`.

Reply:
Done